### PR TITLE
Upgrade gem dependencies for vagrant 1.8.1

### LINKED ISSUE #47: Plugin not working with Vagrant 1.8.1
When installing your plugin (which worked really well in some of my past projects), I have the following issue on my current setup.
Running 'vagrant up' as well as 'vagrant provision' I get the following Error:

    ==> default: Installing Puppet modules in "puppet" with Librarian-Puppet...
    Unable to load puppet. Please install it using native packages for your platform (eg .deb, .rpm, .dmg, etc).
    puppet --version returned pid 5194 exit 1
    /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/defaults.rb:343: warning: duplicated key at line 344 ignored: :queue_type
    /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor/safe_yaml/lib/safe_yaml/syck_node_monkeypatch.rb:42:in `<top (required)>': uninitialized constant Syck (NameError)
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor/safe_yaml/lib/safe_yaml.rb:197:in `require'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor/safe_yaml/lib/safe_yaml.rb:197:in `<module:YAML>'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor/safe_yaml/lib/safe_yaml.rb:132:in `<top (required)>'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor/require_vendored.rb:4:in `require'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor/require_vendored.rb:4:in `<top (required)>'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor.rb:38:in `require'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor.rb:38:in `require_libs'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor.rb:51:in `load_vendored'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet.rb:164:in `<module:Puppet>'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet.rb:29:in `<top (required)>'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/util/command_line.rb:12:in `require'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/util/command_line.rb:12:in `<top (required)>'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/bin/puppet:3:in `require'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/bin/puppet:3:in `<top (required)>'
        from /Users/messner.j/.vagrant.d/gems/bin/puppet:23:in `load'
        from /Users/messner.j/.vagrant.d/gems/bin/puppet:23:in `<main>'

1. **duplicate definition** in ~/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/defaults.rb
This can easily be fixed by removing one of the two similar entries. With unistall and reinstall of this plugin the duplicate will be present again.

    :queue_type => {
      :default    => "stomp",
      :desc       => "Which type of queue to use for asynchronous processing.",
    },

2. **uninitialized constant Syck**
Here I did not find any fix, but I'm no ruby developer and am totally confused by the many different versions of ruby and its gems.
Do I have to install the syck gem? And If so - where?

I have the following environment on my System

    $ vagrant --version
    $ Vagrant 1.8.1

    $ vagrant plugin list
    $ vagrant-librarian-puppet (0.9.1)
    $ vagrant-share (1.1.5, system)

    $ ruby --version
    ruby 2.0.0p481 (2014-05-08 revision 45883) [universal.x86_64-darwin14]

### PR DESCRIPTION
**Pull request by njam**
Closes #47

### CODE REVIEW

File: vagrant-librarian-puppet.gemspec [hunk 1]
```diff
@@ -23,11 +23,12 @@ Gem::Specification.new do |spec|
...
   # which pulls in i18n ~> 0.7,
   # which is incompatible with vagrant 1.7.2 due to a hard dep on i18n 0.6.11.
+  # See https://github.com/rodjek/librarian-puppet/pull/331
   #
   # it's a bit surprising that bundler can't resolve this itself...
   spec.add_runtime_dependency "her", "< 0.7.3"
-  spec.add_runtime_dependency "puppet", "~> 3.4.3"
-  spec.add_development_dependency "bundler", "~> 1.3"
+  spec.add_runtime_dependency "puppet", "~> 4.3.1"
+  spec.add_development_dependency "bundler", ">= 1.5.2", "<= 1.10.6"
```
Comment (jhoblitt) [inline vagrant-librarian-puppet.gemspec]: Why is blunder constrained to be less than 1.10.6?

Comment (njam) [inline vagrant-librarian-puppet.gemspec]: That's vagrant 1.8's constraint, see https://github.com/mitchellh/vagrant/blob/v1.8.1/vagrant.gemspec#L18

Comment (JWGmeligMeyling) [inline vagrant-librarian-puppet.gemspec]: I believe that it's more a "tested and works with 1.10.6" than that problems have been seen with newer versions though. Besides that, this PR works like a charm for me :smile:

**Comment by tuxinaut** [comment]
With these changes / pull request `vagrant-librarian-puppet` runs fine again.

**Comment by stringsonfire** [comment]
Any news on this one? It's been waiting a while

**Comment by tuxinaut** [comment]
@jhoblitt could you merge this pull request? Or are there reasons against it? Because so far we don't have a working out of the box `vagrant-librarian-puppet` version for `vagrant` 1.8.X

**Comment by jhoblitt** [comment]
I went ahead and merged #51 instead of this PR.  We'll pin the bundler version if it proves to be a problem.

**Comment by jhoblitt** [comment]
0.9.2 has been released.

**Pull request closed without merging**